ski_lint/cli.py

```diff
@@ -28,43 +32,62 @@ def get_args() -> Namespace:
     )
     ap.add_argument("filenames", nargs="+", metavar="FILENAME", help="path to the files to check")
     ap.add_argument("--context-width", type=int, default=50, help="width of the context of the non-ASCII line")
+    ap.add_argument("--config", type=str, default=".ski-lint.yml", help="path to config file")
     args = ap.parse_args()
     return args
 
 
-def run(*filenames: str, check: bool = False, context_width: int) -> int:
+def get_config(filename: str) -> dict[str, Any]:
+    config = {}
+    config_file = Path(filename)
+    if config_file.exists():
+        with open(config_file) as yaml_file:
+            config = yaml.safe_load(yaml_file)
+            config = {} if config is None else config
+    return config
+
+
+def run(*filenames: str, check: bool = False, context_width: int, accepted_chars: list[str] = []) -> int:
     bad_encodings = get_non_ascii_files(*filenames)
 
+    has_non_ascii_files = False
+
     if bad_encodings:
         for filename, encoding in bad_encodings.items():
             non_ascii_lines = get_non_ascii_lines(filename)
 
             for line_result in non_ascii_lines:
                 for char, char_positions in line_result.chars.items():
-                    for char_pos in char_positions:
-                        context = extract_context(line_result.line, char_pos, context_width)
-
-                        error_msg = f"{filename} ({encoding}), "
-                        error_msg += f"line {line_result.line_num}, "
-                        error_msg += f"pos {char_pos}, "
-                        if not char.isprintable():
-                            error_msg += f"non-printable char U+{ord(char):0x}, "
-                            context = context.replace(char, f"U+{ord(char):0x}")
-                        else:
-                            error_msg += f"char '{char}', "
-                        error_msg += f"context: '{context}'"
-
-                        log.error(error_msg)
-
+                    if char not in accepted_chars:
+                        has_non_ascii_files = True
+                        for char_pos in char_positions:
+                            context = extract_context(line_result.line, char_pos, context_width)
+
+                            error_msg = f"{filename} ({encoding}), "
+                            error_msg += f"line {line_result.line_num}, "
+                            error_msg += f"pos {char_pos}, "
+                            if not char.isprintable():
+                                error_msg += f"non-printable char U+{ord(char):0x}, "
+                                context = context.replace(char, f"U+{ord(char):0x}")
+                            else:
+                                error_msg += f"char '{char}', "
+                            error_msg += f"context: '{context}'"
+
+                            log.error(error_msg)
+
+    if has_non_ascii_files:
+        if check:
+            return 1
```

Comment:
```suggestion
    if has_non_ascii_files and check:
        return 1
```
won't such construction work ? I seems a bit easier to read to me

Reply:
If you format it like that the check-ok log will also be printed if you have ascii errors, but the check flag is not set. Let me see if I can program it in a more intuitive way